src/modules/interface.js

```diff
@@ -0,0 +1,231 @@
+import ToDo from './todo.js';
+import Storage from './storage.js';
+
+export default class Interface {
+  static toDoList = Storage.getTasks();
+
+  static completedTasks = [];
+
+  static editList = '';
+
+  static displayToDoList = () => {
+    const toDoListContainer = document.querySelector(
+      '.toDoItem',
+    );
+    const clearBtn = document.querySelector('.clearTask');
+    if (Interface.toDoList.length === 0) {
+      clearBtn.classList.add('hide');
+      toDoListContainer.innerHTML = '';
+      toDoListContainer.innerHTML = 'Add A Task At Least!';
+      toDoListContainer.classList.add('errorMsg');
+    } else {
+      toDoListContainer.innerHTML = '';
+      clearBtn.classList.remove('hide');
+      // sort the ToDo array by index
+      Interface.toDoList.sort((a, b) => a.index - b.index);
+      Interface.toDoList.forEach((task) => {
+        const taskIndex = task.index;
+        const taskDescription = task.description;
+        const listItem = document.createElement('div');
+        listItem.className = 'toDoList';
+        listItem.innerHTML = `
+  <input type="checkbox" class="checkbox" id="${taskIndex}" name="${taskDescription}" value="${taskDescription}"><i id="${taskIndex}" class="fa fa-check hide tick" aria-hidden="true"></i><div class='taskDescription'>${taskDescription}</div>
+  <input type="text" class="editInput hide" id="${taskIndex}" name="${taskDescription}" value="${taskDescription}">
+  <button type="button" class="optionBtn" ><i class="fa fa-ellipsis-v fa-2xl menu" id="${taskIndex}" aria-hidden="true"></i></button>
+  <button type="button" class="deleteBtn hide" ><i class="fa fa-trash fa-xl trash-bin" aria-hidden="true"></i></button>
+  `;
+        toDoListContainer.appendChild(listItem);
+      });
+      // add event listener to each ellipses
+      const optionBtn = document.querySelectorAll('.optionBtn');
+      optionBtn.forEach((btn) => {
+        btn.addEventListener('click', (event) => {
+          const listItem = event.target.closest('.toDoList');
+          if (Interface.editList === '') {
+            Interface.editList = listItem;
+            Interface.editTask(Interface.editList);
+          } else if (Interface.editList) {
+            if (Interface.editList !== listItem) {
+              Interface.cancelEditTask(Interface.editList);
+              Interface.editList = listItem;
+              Interface.editTask(Interface.editList);
+            } else if (Interface.editList === listItem) {
+              Interface.cancelEditTask(listItem);
+              Interface.editTask(Interface.editList);
+            }
+          }
+        });
+      });
+
+      // add event listener to each checkbox
+      const checkboxes = document.querySelectorAll('.checkbox');
+      checkboxes.forEach((checkbox) => {
+        checkbox.addEventListener('change', (event) => {
+          Interface.lineOnTask(event.target);
+        });
+      });
```

Comment:
- [ ] As mentioned by the previous reviewer, when I click on the task to change the `done` status from false to true, nothing happens in the local storage.this behaviour doesn't comply with following requirement:
- All changes to the To Do List should be saved in local storage.
 Also user cannot undo the check task, so If user mark the task complete, he cannot undo that. kindly consider fixing these two bugs in your project.